Hogajama/hogajama-domain/src/test/java/com/gepardec/hogarama/testdata/TestDataProducer.java

```diff
@@ -0,0 +1,56 @@
+package com.gepardec.hogarama.testdata;
+
+import static com.gepardec.hogarama.domain.DateUtils.toDate;
+import static com.gepardec.hogarama.domain.DateUtils.toLocalDateTime;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+import java.util.List;
+
+import com.gepardec.hogarama.domain.sensor.SensorData;
+
+public class TestDataProducer {
+
+	private ArrayList<SensorData> data;
+	private SensorData akt;
+	private LocalDateTime lastTime;
+	
+	public TestDataProducer(SensorData sensorData) {
+		data = new ArrayList<SensorData>();
+		akt = sensorData;
+		data.add(akt);
+		lastTime = toLocalDateTime(akt.getTime());
+	}
+
+	public void addValueMinusMinutes(double value, int minutes) {
+		addValueAt( value, lastTime.minus(Duration.ofMinutes(minutes)));
+	}
+
+	public void addValueAt(double value, LocalDateTime time) {
+		akt = copy(akt);
+		akt.setId(Integer.toString(Integer.parseInt(akt.getId()) + 1 ));
+		lastTime = time;
+		akt.setTime(toDate(lastTime));
+		akt.setValue(value);
+		data.add(akt);
+	}
+
+	
+	private SensorData copy(SensorData from) {
```

Comment:
This part of the code is vulnerable to changes in the SensorData class. Maybe we should use a third-party library that enables deep copies of objects